attestation-agent/attester/src/tdx/mod.rs

```diff
@@ -44,22 +83,18 @@ impl Attester for TdxAttester {
 
         report_data.resize(TDX_REPORT_DATA_SIZE, 0);
 
-        let tdx_report_data = tdx_attest_rs::tdx_report_data_t {
-            // report_data.resize() ensures copying report_data to
-            // tdx_attest_rs::tdx_report_data_t cannot panic.
-            d: report_data.as_slice().try_into().unwrap(),
-        };
+        let quote_bytes = TsmReportPath::new(TsmReportProvider::Tdx).map_or_else(
+            |notsm| {
+                get_quote_ioctl(&report_data).map_err(|e| anyhow!("TDX Attester: {}{}", e, notsm))
```

Comment:
I think you want `.context` here rather than `map_err` since the error you're already getting an `Anyhow::result`

Reply:
fixed